aten/src/TH/THAllocator.c

```diff
@@ -126,11 +137,125 @@ ptrdiff_t THMapAllocatorContext_size(THMapAllocatorContext *ctx)
 
 void THMapAllocatorContext_free(THMapAllocatorContext *ctx)
 {
-  if (ctx->filename != unknown_filename)
+  if (ctx->filename != unknown_filename) {
     THFree(ctx->filename);
+#ifdef _WIN32
+    THFree(ctx->eventname);
+#endif
+  }
   THFree(ctx);
 }
 
+#ifdef _WIN32
+typedef struct {
+  HANDLE event[MAXIMUM_WAIT_OBJECTS];
+  HANDLE handle[MAXIMUM_WAIT_OBJECTS];
+  void *data[MAXIMUM_WAIT_OBJECTS];
+  int nCount;
+  CRITICAL_SECTION* CriticalSection;
+  HANDLE event_process;
+} ReleaseContextList;
+static ReleaseContextList* context_list;
+static void AddToList(ReleaseContextList* context_list, HANDLE event, HANDLE handle, void* data) {
+  EnterCriticalSection(context_list->CriticalSection);
```

Comment:
We are blocking the main thread using Critical Section. Hope there is a better solution.